schema_enforcer/schemas/validator.py

```diff
@@ -0,0 +1,110 @@
+"""Classes for custom validator plugins."""
+# pylint: disable=no-member, too-few-public-methods
+# See PEP585 (https://www.python.org/dev/peps/pep-0585/)
+from __future__ import annotations
+import pkgutil
+import inspect
+import jmespath
+from schema_enforcer.validation import ValidationResult
+
+
+class BaseValidation:
+    """Base class for Validation classes."""
+
+    def __init__(self):
+        """Base init for all validation classes."""
+        self._results: list[ValidationResult] = []
+
+    def add_validation_error(self, message: str, **kwargs):
+        """Add validator error to results.
+
+        Args:
+          message (str): error message
+          kwargs (optional): additional arguments to add to ValidationResult when required
+        """
+        self._results.append(ValidationResult(result="FAIL", schema_id=self.id, message=message, **kwargs))
+
+    def add_validation_pass(self, **kwargs):
+        """Add validator pass to results.
+
+        Args:
+          kwargs (optional): additional arguments to add to ValidationResult when required
+        """
+        self._results.append(ValidationResult(result="PASS", schema_id=self.id, **kwargs))
+
+    def get_results(self) -> list[ValidationResult]:
+        """Return all validation results for this validator."""
+        if not self._results:
+            self._results.append(ValidationResult(result="PASS", schema_id=self.id))
```

Comment:
Somewhat adjacent to #46, If there are no validation results, do we want to print a log message indicating that nothing was validated?

Reply:
May defer to @dgarros here. I assumed the intent was to support the use case of no failure is equal to a pass, but that doesn't really fit the current runtime model.